Added PXE BOOT client to the vagrant setup

jirihnidek opened the pull request:
* It is possible to create new VM booting from PXE server
* You have to create PXE BOOT server first using:

    vagrant up pxe-server

* When PXE BOOT server is installed and running, then you can
  create another VM booting from this PXE server using

    vagrant up pxe-client

* The client will be kickstarted and Anaconda (GUI installer)
  will start boot process, because all required settings are
  present in ks.cfg configuration file.
* The client is halted at the end of installation process,
  but vagrant will still wait for SSH to become available.

jirihnidek commented:
This PR is blocked by #1784. I will rebase this PR, when the 1784 will be merged.

kahowell commented on Vagrantfile:
```diff
@@ -124,7 +125,27 @@ Vagrant.configure("2") do |config|
+  config.vm.define 'pxe-client'
+  config.ssh.username = 'root'
+
+  config.vm.network "private_network",
```
Since this network definition occurs twice, it seems to cause vagrant-libvirt to define two NICs on the pxe-server0 network...

jirihnidek replied:
Good point.

kahowell commented on Vagrantfile:
```diff
@@ -124,7 +125,27 @@ Vagrant.configure("2") do |config|
+    libvirt__dhcp_bootp_server: ip_addr_server,
+    libvirt__network_name: 'pxe-server0'
+
+  config.vm.provider :libvirt do |domain|
```
This asks vagrant-libvirt to apply these settings to all the VMs... instead we should specify this in the host-specific way. (see other comments).

jirihnidek commented:
@kahowell Thanks for you suggestion with one `Vagrant.configure` block. I hope I addressed all your requests. I also rebased this branch against current `vagrant_pxeserver` branch.
